cubano-concordion/src/main/java/org/concordion/cubano/framework/ConcordionBase.java

```diff
@@ -103,6 +111,30 @@ public Browser getBrowser(String key, BrowserProvider browserProvider) {
         return browsers.get(key);
     }
 
+    private void incrementBrowserTestCount() {
+        if (browserCloseAfterXTests <= 0 || browserTestRunCounted) {
+            return;
+        }
+
+        browserTestRunCounted = true;
+        Integer count = testCount.get();
+
+        if (count >= browserCloseAfterXTests) {
+            Map<String, Browser> browsers = threadBrowsers.get();
+
+            for (Iterator<String> iterator = browsers.keySet().iterator(); iterator.hasNext();) {
+                Browser browser = browsers.get(iterator.next());
+
+                browser.close();
```

Comment:
This could close the other thread's browsers while they are in use. Is suggest only closing this thread's browser if possible.

Reply:
See above comments, this should be a list of browsers for the current thread (in the event they have more than one browser open for the thread), not all browsers across all threads.